### PR TITLE
MessageAggregator Potential Leak

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
MessageAggregator has a potential to leak if a new message is received before the existing message has completed, and if a HttpContent is received but maxContentLength has been exceeded, or the content length is too long.

Modifications:

- Make the HttpObjectAggregator more robust to leaks
- Reduce dependance on handlingOversizedMessage but instead rely on the more general check of a null currentMessage

Result:
More robust MessageAggregator with less chance of leaks

### CODE REVIEW

**Comment by Scottmitch** [comment]
This is the 4.1 version of https://github.com/netty/netty/pull/5676

**Comment by Scottmitch** [comment]
@normanmaurer - PTAL

**Comment by normanmaurer** [comment]
@Scottmitch still want to merge this one ?

**Comment by Scottmitch** [comment]
@normanmaurer - Yes. Same reasons as https://github.com/netty/netty/pull/5676#issuecomment-246879433. WDYT?

File: codec/src/main/java/io/netty/handler/codec/MessageAggregator.java [hunk 1]
```diff
@@ -174,6 +175,7 @@ public final void setMaxCumulationBufferComponents(int maxCumulationBufferCompon
         }
     }
 
+    @Deprecated
```
Comment (normanmaurer) [inline codec/src/main/java/io/netty/handler/codec/MessageAggregator.java]: @Scottmitch also adding some javadocs and just tell that it will go away ?

**Comment by normanmaurer** [comment]
@Scottmitch one nit.

File: codec/src/main/java/io/netty/handler/codec/MessageAggregator.java (hunk 1 above)
Comment (Scottmitch) [inline codec/src/main/java/io/netty/handler/codec/MessageAggregator.java]: +1

**Comment by Scottmitch** [comment]
after build completes I'll ship it

**Comment by Scottmitch** [comment]
4.1 (e3462a7)

**Pull request closed without merging**